qttypes/src/lib.rs

```diff
@@ -1734,6 +1734,46 @@ impl fmt::Debug for QStringList {
     }
 }
 
+impl<T, const N: usize> From<[T; N]> for QStringList
+where
+    QString: From<T>,
+{
+    fn from(s: [T; N]) -> Self {
+        let mut list = QStringList::new();
+        for i in s {
+            list.push(QString::from(i));
+        }
+        list
+    }
+}
+
+impl<T> From<Vec<T>> for QStringList
```

Comment:
should it maybe be `From<&[T]>`  ?

Reply:
Well, it can be done but since `QString`, `String` do not implement Copy, the new impl will look like this:
```rust
impl<T> From<&[T]> for QStringList
where
    T: Clone,
    QString: From<T>,
{
    fn from(s: &[T]) -> Self {
        let mut list = QStringList::new();
        for i in s {
            list.push(QString::from(i.clone()));
        }
        list
    }
}
```
There might be a way to implement this without clone, but I'm not sure how to do it. The vector implementation alone does not use any clone. So I can add the slice implementation too, but I don't think it should replace the vector implementation, at least not before we can get rid of all the cloning.

Reply:
Cloning a QString is cheap (just increment a reference count)
But creating a Vec is not cheap (allocation)
If you have a `Vec<T>` you can get a `&[T]` for free. But if you have a `&[T]`  you need to allocate to create a Vec.

Reply:
Hmm. So should I remove the Vec implementation or just add the slice implementation leaving the Vec one intact?